core/include/scipp/core/element/special_values.h

```diff
@@ -12,6 +12,58 @@
 
 namespace scipp::core::element {
 
+constexpr auto isnan =
+    overloaded{arg_list<double, float>,
+               [](const auto x) {
+                 using std::isnan;
+                 return isnan(x);
+               },
+               [](const units::Unit &) { return units::dimensionless; }};
+
+constexpr auto isinf =
+    overloaded{arg_list<double, float>,
+               [](const auto x) {
+                 using std::isinf;
+                 return isinf(x);
+               },
+               [](const units::Unit &) { return units::dimensionless; }};
+
+constexpr auto isfinite =
+    overloaded{arg_list<double, float>,
+               [](const auto x) {
+                 using std::isfinite;
+                 return isfinite(x);
+               },
+               [](const units::Unit &) { return units::dimensionless; }};
+
+namespace detail {
+template <typename T>
+auto isposinf(T x) -> std::enable_if_t<std::is_floating_point_v<T>, bool> {
+  return std::isinf(x) && !std::signbit(x);
+}
+
+template <typename T>
+auto isneginf(T x) -> std::enable_if_t<std::is_floating_point_v<T>, bool> {
```

Comment:
Why is the SFINAE required?

Reply:
Not sure if it really is required. At the moment, this is mainly there to disambiguate calls to `isposinf(ValueAndVariance<T>)`. But since there is no `std::isposinf`, I wanted to have an implementation specifically for floating point numbers in order to use ADL as usual in the kernels.